compiler-rt/lib/hwasan/hwasan_linux.cpp

```diff
@@ -106,8 +106,13 @@ static uptr GetHighMemEnd() {
 }
 
 static void InitializeShadowBaseAddress(uptr shadow_size_bytes) {
-  __hwasan_shadow_memory_dynamic_address =
-      FindDynamicShadowStart(shadow_size_bytes);
+  // Android static HWASan doesn't initialize flags correctly
+  if (!SANITIZER_ANDROID && flags()->fixed_shadow_base != (uptr)-1) {
```

Comment:
please use SANITIZER_CAN_USE_PREINIT_ARRAY
as it's the real rootcause

Reply:
Done